Source/missiles.cpp

```diff
@@ -4194,11 +4120,11 @@ void MI_Rportal(Missile &missile)
 
 static void DeleteMissiles()
 {
-	for (int i = 0; i < ActiveMissileCount;) {
-		if (Missiles[ActiveMissiles[i]]._miDelFlag) {
-			DeleteMissile(i);
+	for (auto it = Missiles.begin(); it != Missiles.end();) {
+		if (it->second._miDelFlag) {
+			it = Missiles.erase(it);
 		} else {
-			i++;
+			++it;
 		}
 	}

```

Comment:
This could use `std::remove_if(Missiles.begin(), Missiles.end(), [](Missile &it) { return it._miDelFlag });` (change the lambda as needed based on the container type 😄 )

Reply:
`std::remove_if` [doesn't work](https://stackoverflow.com/questions/9210014/remove-elements-from-an-unordered-map-fulfilling-a-predicate) with `unordered_map`. 😢
And I'm not sure if [`erase_if`](https://en.cppreference.com/w/cpp/container/unordered_map/erase_if) is supported on all platforms (c++ 20 needed).

Reply:
Didn't realise the implementation of `std::remove_if` uses move/resize 😓 . This would require a different datatype to make use of that helper. The two list types provide their own specialisation of `remove_if` which would be suitable if that container gets the pick.

Reply:
I changed to `remove_if` on `std::list`.